flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaSerializerWrapperTest.java

```diff
@@ -0,0 +1,63 @@
+package org.apache.flink.connector.kafka.sink;
+
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.metrics.groups.UnregisteredMetricsGroup;
+import org.apache.flink.util.FlinkUserCodeClassLoaders;
+import org.apache.flink.util.SimpleUserCodeClassLoader;
+import org.apache.flink.util.UserCodeClassLoader;
+
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.Test;
+
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests for {@link KafkaSerializerWrapper}. */
+public class KafkaSerializerWrapperTest {
+    @Test
+    public void testUserCodeClassLoaderIsUsed() throws Exception {
+        final KafkaSerializerWrapperCaptureForTest wrapper =
+                new KafkaSerializerWrapperCaptureForTest();
+        final ClassLoader classLoader =
+                FlinkUserCodeClassLoaders.childFirst(
+                        new URL[0],
+                        getClass().getClassLoader(),
+                        new String[0],
+                        throwable -> {},
+                        true);
+        wrapper.open(
+                new SerializationSchema.InitializationContext() {
+                    @Override
+                    public MetricGroup getMetricGroup() {
+                        return new UnregisteredMetricsGroup();
+                    }
+
+                    @Override
+                    public UserCodeClassLoader getUserCodeClassLoader() {
+                        return SimpleUserCodeClassLoader.create(classLoader);
+                    }
+                });
+
+        assertEquals(classLoader, wrapper.getClassLoaderUsed());
+    }
```

Comment:
Could you please add another test case for previous classloader? With a descriptive test similar to yours, e.g, `testDefaultClassLoaderIsUsed`.

In this case you can use the `DummyInitializationContext`.

The test `KafkaRecordDeserializationSchemaTest` already tests this but the assertion is not on the classloader, so having default classloader assertion (e.g, `DummyInitializationContext.class.getClassLoader()`) make it clearer.

Reply:
Same comment for the deserialization test below

Reply:
Sure. Just added a couple tests.